File: js/awqat/token.js
import { readFileSync, writeFileSync } from 'node:fs';

// Token önbelleği: login/refresh sonrası token'ın diske yazılan kalıcı hâli.
// Sonraki çalıştırmalarda geçerliyse yeniden login ATILMAZ (kota korunur).
// Biçim Go ile UYUMLU: { accessToken, refreshToken, expiry (ISO 8601) }.

export function loadTokenCache(path) {
  if (!path) return null;
  try {
    const tc = JSON.parse(readFileSync(path, 'utf8'));
    return tc && tc.accessToken ? tc : null;
  } catch {
    return null;
  }
}

// Token'ı yalnızca sahibin okuyabileceği (0600) bir dosyaya yazar.
// Hata sessizce yutulur — önbellek bir kolaylıktır, kritik değildir.
export function saveTokenCache(path, tc) {
  if (!path) return;
  try {
    writeFileSync(path, JSON.stringify(tc, null, 2), { mode: 0o600 });
  } catch {
    // yok say
  }
}

File: python/awqat/token.py
"""Token önbelleği: login/refresh sonrası token'ı diske kalıcı yazar.

Sonraki çalıştırmalarda geçerliyse yeniden login ATILMAZ (kota korunur).
Biçim Go/JS ile UYUMLU: {"accessToken", "refreshToken", "expiry" (ISO 8601)}.
"""

import json
import os


def load_token_cache(path):
    if not path:
        return None
    try:
        with open(path, "r", encoding="utf-8") as f:
            tc = json.load(f)
        return tc if tc.get("accessToken") else None
    except (OSError, ValueError):
        return None


def save_token_cache(path, tc):
    """Token'ı yalnızca sahibin okuyabileceği (0600) dosyaya yazar. Hata yutulur."""
    if not path:
        return
    try:
        with open(path, "w", encoding="utf-8") as f:
            json.dump(tc, f, indent=2)
        os.chmod(path, 0o600)
    except OSError:
        pass
